Rakefile

```diff
@@ -1,24 +1,19 @@
 require 'rake'
 require 'rake/testtask'
 
+require_relative 'lib/tasks/exercise'
+require_relative 'lib/tasks/exercise_test_tasks'
 require_relative 'lib/tasks/exercise_tests_runner'
-require_relative 'lib/tasks/knows_what_exercises_are_called_and_where_they_live_in_the_directory_structure'
+require_relative 'lib/tasks/rake_arguments'
 
-Exercise.all.each do |exercise|
-  task exercise do
-    ExerciseTestsRunner.new(
-      exercise: exercise,
-      test_options: ARGV.drop_while { |e| e != '--' }.drop(1).join(' ')
-    ).run
-  end
-end
+ExerciseTestTasks.new exercises: Exercise.all,
+                      options: RakeArguments.task_options
 
 task default: 'test'
 
 desc 'Run individual exercises or run all development and exercise tests'
 task :test do
-  exercises = ARGV.take_while { |e| e != '--' }.drop(1)
-  next if exercises.any?
+  next if RakeArguments.more_tasks?
```

Comment:
I'm beginning tho think this workaround has too many problems to be the right solution.

For example
```
rake test --backtrace
```

Will not run any tests because there are more "tasks". This could be fixed by checking the next task agains the list of tasks, but I think this is really just a bandaid on the fact that rake is really not made to work this way.

Personally, I think it would be preferable to have `rake test` **always** run all tests, and namespace the other tests, so they would be `rake test:hamming`. This makes calling multiple exercise tests at once a bit more wordy(`rake test:hamming test:series`), but I'm not sure that's a thing that happens too much.